src/Workspaces/Core/Portable/EmbeddedLanguages/RegularExpressions/RegexLexer.cs

```diff
@@ -0,0 +1,448 @@
+        {
+            var trivia = ScanLeadingTrivia(allowTrivia, options);
+            if (Position == Text.Length)
+            {
+                return CreateToken(RegexKind.EndOfFile, trivia, ImmutableArray<VirtualChar>.Empty);
+            }
+
+            var ch = this.CurrentChar;
+            Position++;
+
+            return CreateToken(GetKind(ch), trivia, ImmutableArray.Create(ch));
+        }
+
+        private static RegexKind GetKind(char ch)
+        {
+            switch (ch)
+            {
+                case '|': return RegexKind.BarToken;
+                case '*': return RegexKind.AsteriskToken;
+                case '+': return RegexKind.PlusToken;
+                case '?': return RegexKind.QuestionToken;
+                case '{': return RegexKind.OpenBraceToken;
+                case '}': return RegexKind.CloseBraceToken;
+                case '\\': return RegexKind.BackslashToken;
+                case '[': return RegexKind.OpenBracketToken;
+                case ']': return RegexKind.CloseBracketToken;
+                case '.': return RegexKind.DotToken;
+                case '^': return RegexKind.CaretToken;
+                case '$': return RegexKind.DollarToken;
+                case '(': return RegexKind.OpenParenToken;
+                case ')': return RegexKind.CloseParenToken;
+                case ',': return RegexKind.CommaToken;
+                case ':': return RegexKind.ColonToken;
+                case '=': return RegexKind.EqualsToken;
+                case '!': return RegexKind.ExclamationToken;
+                case '<': return RegexKind.LessThanToken;
+                case '>': return RegexKind.GreaterThanToken;
+                case '-': return RegexKind.MinusToken;
+                case '\'': return RegexKind.SingleQuoteToken;
+                default: return RegexKind.TextToken;
+            }
+        }
+
+        private ImmutableArray<RegexTrivia> ScanLeadingTrivia(bool allowTrivia, RegexOptions options)
+        {
+            if (!allowTrivia)
+            {
+                return ImmutableArray<RegexTrivia>.Empty;
+            }
+
+            var result = ArrayBuilder<RegexTrivia>.GetInstance();
+
+            var start = Position;
+
+            while (Position < Text.Length)
+            {
+                var comment = ScanComment(options);
+                if (comment != null)
+                {
+                    result.Add(comment.Value);
+                    continue;
+                }
+
+                var whitespace = ScanWhitespace(options);
+                if (whitespace != null)
+                {
+                    result.Add(whitespace.Value);
+                    continue;
+                }
+
+                break;
+            }
+
+            return result.ToImmutableAndFree();
+        }
+
+        public RegexTrivia? ScanComment(RegexOptions options)
+        {
+            if (Position < Text.Length)
```

Comment:
Nit: Perhaps invert `if` and bail out early?

Reply:
I liked there being only one place where null was returned :)   I could cosnider tweaking this though.

Reply:
#resolved